parser/src/format/character_separated/w3c_extended_log.rs

```diff
@@ -0,0 +1,181 @@
+//! The [W3C Extended Log format](https://www.w3.org/TR/WD-logfile.html) is a made specifically for
+//! http server logs. It's essentially just a tab-separated values file, with a few extra
+//! differences. The main thing being that the column headers appear in a special `#Fields`
+//! directive instead just being on the first row. Also, nulls are represented as `-`.
+use super::{resolve_headers, CsvOutput};
+use crate::config::ParseConfig;
+use crate::format::projection::build_projections;
+use crate::format::{Output, ParseError, Parser};
+use crate::input::Input;
+use std::io::BufRead;
+
+// TODO: Put a span in the error so we can display the source.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
```

Comment:
When I added that, I was working with a different approach, where I was using `nom` to parse the headers, which would have made it easy to put span info here. But it ended up being more involved than seemed worth it. I just removed this todo.

Generally, though, it might be nice to someday revisit parser error handling to include spans where possible. I came across the [miette project](https://github.com/zkat/miette) a little while ago, which seemed like a nice way to do that.